edkrepo_manifest_parser/edk_manifest.py

```diff
@@ -378,7 +394,34 @@ def get_combo_element(self, name):
         for combo in combinations.iter(tag='Combination'):
             if combo.attrib['name'] == name:
                 return copy.deepcopy(combo)
-        raise ValueError(COMB_UNKOWN_ERROR.format(name, self._fileref))
+        raise ValueError(COMB_UNKNOWN_ERROR.format(name, self._fileref))
+
+    def get_all_patchsets(self):
+        patchsets = []
+        for patch in self._patch_sets.keys():
+            patchsets.append(self._patch_sets[patch])
+        return patchsets
+
+    def get_parent_patchset_operations(self, name, patch_set_operations):
+        parent_sha = self._patch_sets[name][2]
+        if parent_sha in self._patch_sets:
+            if self._patch_sets[parent_sha][0] == self._patch_sets[name][0]:
+                self.get_parent_patchset_operations(parent_sha, patch_set_operations)
+                patch_set_operations.append(self._patch_set_operations[parent_sha])
+            else:
+                raise ValueError(REMOTE_DIFFERENT_ERROR.format(parent_sha, self._patch_sets[parent_sha][0], 
+                name, self._patch_sets[name][0]))
+
+    def get_patchset_operations(self, name=None):

```

Comment:
please add a doc string to this method describing the input parameters and the format and content of the return value. In this case a brief description of the algorithm to calculate the data would also be appropriate.